src/main/scala/karazin/scala/users/group/week1/homework/Homework.scala

```diff
@@ -36,94 +37,91 @@ import scala.annotation.tailrec
  * For more details @see https://en.wikipedia.org/wiki/Kolakoski_sequence
  */
 
-object Homework :
+object Homework:
 
-  object `Boolean Operators` :
+  object `Boolean Operators`:
 
 def not(b: Boolean): Boolean =
   if b then false
   else true
 
 def and(left: Boolean, right: Boolean): Boolean =
-  if not(left) then false
-  else if not(right) then false
-  else true
+  if left then right
+  else false
 
 def or(left: Boolean, right: Boolean): Boolean =
```

Comment:
`and` must be eager and works as `&&` means `right` should not be calculated if `left` is `false`. This and fails on `and(false, loop)`. Please review the first lecture on Scala.
When you fix it then add additional test to check it.
Hint:

since all arguments are well-known, then use `propBoolean` instead of `forAll`;
`throw` exception for right argument;
provide meaningful error message for the thrown exception.

Please apply all this comments for `or`.

Reply:
property("and") = forAll { (pair: (Boolean, Boolean)) =>
    val (left, right) = pair

    and(left,right)== (left && right)
  }

  property("and")=propBoolean{
    val (left, right) = (false,loop)

    and(left, right) == (left && right)

  }

  property("or") = forAll { (pair: (Boolean, Boolean)) =>
    val (left, right) = pair

    or(left,right) == (left || right)
  }
  property("or") = propBoolean {
    val (left, right) = (true,loop)

    or(left, right) == (left || right)
  }
???